selftests/unit/test_archive.py

```diff
@@ -125,9 +125,11 @@ def get_path(*args):
         self.assertEqual(os.path.realpath(get_path("link_to_dir")),
                          get_path("dir"))
         # File permissions
-        # TODO: Handle permission correctly for all users
-        # Default perm by user is 0o664 and by root 0o644
-        default_perm = 0o664 if os.getuid() else 0o644
+        # Different user and/or systems have different default permissions.
```

Comment:
Wouldn't things be more predictable (and have less code) with `os.umask()`? I don't anticipate a problem if we set a default umask to be used during the tests.